### PR TITLE
Cross compiles project to Scala 3

### PR DESCRIPTION
**Pull request by amumurst**
This PR contains a full reimplementation of all macros for Dotty. 

This is the first time I write any macros at all for scala, so its ***highly*** likely that they are somewhere on the scale from `completely wrong` to `somewhat okay` 😄  

Else there are some slight changes to the test classes to get Dotty running. 

The "API" of the loggers is the same, but I have absolutely no clue if things are breaking or not 😅

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -9,3 +9,5 @@ addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.13")
 addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.9.5")
 
 addSbtPlugin("com.scalapenos" % "sbt-prompt" % "1.0.2")
+
+addSbtPlugin("ch.epfl.lamp" % "sbt-dotty" % "0.4.1")
```
Comment (sullis) [inline project/plugins.sbt:13]: please upgrade to sbt-dotty 0.5.1

File: project/Dependencies.scala [hunk 2]
```diff
@@ -2,16 +2,17 @@ import sbt._
 
 object Version {
   val logback   = "1.2.3"
-  val mockito   = "1.5.17"
+  val mockito   = "3.2.1.0"
+  val dotty     = "0.26.0-RC1"
```
Comment (sullis) [inline project/Dependencies.scala:6]: upgrade to 3.0.0-M3

File: .travis.yml [hunk 3]
```diff
@@ -5,6 +5,7 @@ scala:
    - 2.12.11
    - 2.11.12
    - 2.13.1
+   - 0.26.0-RC1
```
Comment (sullis) [inline .travis.yml:8]: upgrade to 3.0.0-M3

**Review by sullis — CHANGES_REQUESTED** [review]
let's upgrade to 3.0.0-M3

**Comment by SethTisue** [comment]
Hi. I have merge rights in this repo now, and I'll be happy to merge this and pull publishing levers if it is updated for Scala 3.0.0-M3.

**Comment by SethTisue** [comment]
See #225 for discussion about the possible future of this repository.

**Comment by SethTisue** [comment]
(needs rebase)

**Comment by SethTisue** [comment]
Oh, and Scala 3.0.0-RC1 just hit Maven Central yesterday!

**Comment by amumurst** [comment]
Yeah! It compiles for RC1 but scalatest stuff is not out yet, probably can wait for those. Working on rebase atm.

**Comment by SethTisue** [comment]
@amumurst (I think it would be fine to land this for M3, then separately land an RC1 upgrade)

**Comment by SethTisue** [comment]
I've requested that the ScalaTest folks publish for RC1: https://github.com/scalatest/scalatest/issues/1138#issuecomment-780718244

**Comment by amumurst** [comment]
@bbonanno I had to kinda revert back to `scalatestplus-mockito` from `mockito-scala` since from what I kan see it is not doing anything about scala 3 so far (https://github.com/mockito/mockito-scala/pull/330.)

Syntax-wise its just replacing imports and `*` to `any`, but I understood from https://github.com/lightbend/scala-logging/pull/255 that the scala version is safer to use.

**Comment by bbonanno** [comment]
@amumurst sadly I'm blocked by a ton of dependencies to progress to support Scala 3, in the same way you were by mockito-scala, so I totally feel your pain.
Yes, it's safer, but I guess the project can do without it until I can add Scala 3 support.

The PR had also changes to tests that were broken regardless of the mockito flavour, so I guess you wanna keep those changes

**Comment by amumurst** [comment]
@bbonanno Totally! 
All changes you did are kept and working, only this https://github.com/lightbend/scala-logging/pull/227/files#diff-33835318a9975954275bb9500a41c274042970ee83606cf05c2e41a8c6bc6616R31 had to be changed in the actual test implementations 👍

**Review by SethTisue — APPROVED** [review]
build changes LGTM. (I'm unable to review the code changes.)

**Comment by amumurst** [comment]
I do not think I have much more to add. Tests are green so it can't be that bad, no? 🤣 I guess @sullis ?

I think if anyone who actually knows these macros should stumble upon this i think these two methods are where there probably are better ways.


https://github.com/lightbend/scala-logging/pull/227/files#diff-0e17135facf79fb676040c8f9c4194ec443f5f2ffe81bc91f7655269fed1f9adR230

replacing
https://github.com/lightbend/scala-logging/blob/0057744a001c5757f5f6976c37ea6743ceace2d8/src/main/scala-2.x/com/typesafe/scalalogging/LoggerMacro.scala#L261

and

https://github.com/lightbend/scala-logging/pull/227/files#diff-0e17135facf79fb676040c8f9c4194ec443f5f2ffe81bc91f7655269fed1f9adR264

replacing
https://github.com/lightbend/scala-logging/blob/0057744a001c5757f5f6976c37ea6743ceace2d8/src/main/scala-2.x/com/typesafe/scalalogging/LoggerMacro.scala#L295

**Comment by SethTisue** [comment]
I'm happy to merge regardless, if it comes to that, but let's let it sit until next week to see if any reviewers appear.

File: project/Dependencies.scala [hunk 4]
```diff
@@ -2,27 +2,29 @@ import sbt._
 
 object Version {
   val logback   = "1.2.3"
-  val mockito   = "1.16.23"
-  val scalaTest = "3.2.3"
+  val mockito   = "3.2.4.0-M1"
+  val scalaTest = "3.2.4-M1"
```
Comment (sullis) [inline project/Dependencies.scala:6]: ScalaTest 3.2.4 is available in Maven Central
https://repo1.maven.org/maven2/org/scalatest/scalatest_3.0.0-RC1/3.2.4/

File: .travis.yml [hunk 5]
```diff
@@ -6,7 +6,7 @@ scala:
   - 2.11.12
   - 2.12.13
   - 2.13.4
-
+  - 3.0.0-M3
```
Comment (sullis) [inline .travis.yml:9]: perhaps this should be 3.0.0-RC1

**Comment by SethTisue** [comment]
👏

**Pull request merged**